pipeline/sj_luigi_luigi.py

```diff
@@ -0,0 +1,191 @@
+        msg = EmailMessage()
+        msg["Subject"] = self.subject
+        msg["From"] = email().email_from
+        msg["To"] = ", ".join(self.receipients)
+        msg.set_content(dedent(self.message))
+
+        # attach files
+        files = self.attach_files
+        for file in files:
+            with open(file, "rb") as f:
+                file_data = f.read()
+                file_name = os.path.basename("pipeline/output.csv")
+
+            msg.add_attachment(file_data, maintype="application", subtype="octet-stream", filename=file_name)
+
+        # send email
+        try:
+            s = smtplib.SMTP_SSL(email().email_server, email().email_out_port)
+            s.login(email().email_from, email().email_password)
+            s.send_message(msg)
+        except Exception as e:
+            print(f"Exception: {e}")
+        finally:
+            print(f"Email sent to {self.receipients}")
+            s.quit()
+
+
+if __name__ == "__main__":
+    luigi.build([LoadToDB(csv_file="data/trip.csv", table_name="trip")], 
+                local_scheduler=True)
+
+    luigi.build([LoadToDB(csv_file="data/zone.csv", table_name="zone")], 
+                local_scheduler=True)
+
+    query = '''
+                SELECT tpep_pickup_datetime, tpep_dropoff_datetime, passenger_count, total_amount, trip_distance, TIMESTAMPDIFF(MINUTE, tpep_pickup_datetime, tpep_dropoff_datetime) AS trip_duration
+                FROM (
+                    SELECT tpep_pickup_datetime, tpep_dropoff_datetime, passenger_count, total_amount, trip_distance, a.Zone AS PULocationZone, b.Zone AS DOLocationZone
+                    FROM `trip`
+                    JOIN `zone` a ON trip.PULocationID = a.LocationID
+                    JOIN `zone` b ON trip.DOLocationID = b.LocationID
+                    ) a
+                WHERE PULocationZone = 'Upper East Side North' AND DOLocationZone = 'Upper East Side South';
+            '''
+
+    luigi.build([QueryDBToCSV(query=query)],
+                local_scheduler=True)
+    
+    output_df = QueryDBToCSV(query).run()
+
+    newline = "\n"
+    message = f'''
+Dear All,
+
+I found something,
+
+From Upper East Side North -> Upper East Side South:
+Number of rows selected: {output_df.shape[0]}
+Average passenger count per trip: {round(sum(output_df.passenger_count) / len(output_df.passenger_count), 2)}
+Average total amount per trip: {round(sum(output_df.total_amount) / len(output_df.total_amount), 2)}
+Average distance per trip: {round(sum(output_df.trip_distance) / len(output_df.trip_distance), 2)}
+Average duration per trip (in minutes): {round(sum(output_df.trip_duration) / len(output_df.trip_duration), 2)}
+
+-------------------------------------------
+Summary (First 5 rows order by pickup time)
+-------------------------------------------
+{'|'.join(output_df.columns.to_list())}
+{newline.join(
+    list('|'.join(map(
+        str, output_df.values.tolist()[:5][i])) 
+        for i in range(5)))}
+'''
```

Comment:
I know you want to solve the indentation. But this definitely not the right way.
```suggestion
    message = dedent(f'''
        Dear All,

        I found something,

        From Upper East Side North -> Upper East Side South:
        Number of rows selected: {output_df.shape[0]}
        Average passenger count per trip: {round(sum(output_df.passenger_count) / len(output_df.passenger_count), 2)}
        Average total amount per trip: {round(sum(output_df.total_amount) / len(output_df.total_amount), 2)}
        Average distance per trip: {round(sum(output_df.trip_distance) / len(output_df.trip_distance), 2)}
        Average duration per trip (in minutes): {round(sum(output_df.trip_duration) / len(output_df.trip_duration), 2)}

        -------------------------------------------
        Summary (First 5 rows order by pickup time)
        -------------------------------------------
        {'|'.join(output_df.columns.to_list())}
    ''')
    for i in range(5):
        message += '|'.join(map(str, output_df.values.tolist()
                            [:5][i])) + newline
```